scip-java-proto/src/main/protobuf/scip.proto

```diff
@@ -191,6 +191,23 @@ message Relationship {
     bool is_implementation = 3;
     // Similar to `references_symbols` but for "Go to type definition".
     bool is_type_definition = 4;
+    // Allows overriding the behavior of "Go to definition" and "Find references"
+    // for symbols which do not have a definition of their own or could
+    // potentially have multiple definitions.
+    //
+    // For example, in a language with single inheritance and no field overriding,
+    // inherited fields can reuse the same symbol as the ancestor which declares
+    // the field. In such a situation, is_definition is not needed.
+    //
+    // On the other hand, in languages with single inheritance and some form
+    // of mixins, you can use is_definition to relate the symbol to the
+    // matching symbol in ancestor classes, and is_reference to relate the
+    // symbol to the matching symbol in mixins.
+    //
+    // NOTE: At the moment, due to limitations of the SCIP to LSIF conversion,
+    // only global symbols in an index are allowed to use is_definition.
+    // The relationship may not get recorded if either symbol is local.
+    bool is_definition = 5;
```

Comment:
This isn't strictly needed for this PR but it will be needed in the next PR and having it here allows `ScipPrinters` to support it off the bat.